<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>571</width>
    <height>720</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>571</width>
    <height>720</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>571</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_Form">
   <item>
    <widget class="QLabel" name="label_note">
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">margin-top: 10px; margin-bottom: 15px;</string>
     </property>
     <property name="text">
      <string>(1) Note: restarting the app may be necessary for certain settings to take effect.
(2) Note: move your cursor over the labels to see an explanation of the settings.
(1) Warning: closing this panel will apply changes and reset the current state automatically.</string>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_general">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>32</height>
      </size>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>General</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="checkBox_setting_disable_selection_cart">
     <property name="toolTip">
      <string>Whether or not to disable selection cart.</string>
     </property>
     <property name="text">
      <string>Disable selection cart</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="checkBox_setting_auto_merge_group">
     <property name="toolTip">
      <string>If checked, selecting a group from the tree view will merge all its children into a single, larger dataframe, allowing for immediate visualization or analysis. This option is only applicable when all children are either scalars or 1D arrays, resulting in a merged dataframe of either 1D or 2D format.</string>
     </property>
     <property name="text">
      <string>Automatically merge dataframes at group level</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_min_big_data_sample_size">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_frame_min_big_data_sample_size">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="label_min_big_data_sample_size">
        <property name="toolTip">
         <string>This is the minimum dataset size required for it to be considered as big data. When in big data mode, data is loaded and visualized using a combination of chunking, downsampling, and caching techniques.</string>
        </property>
        <property name="text">
         <string>Minimum sample size for big data  </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBox_min_big_data_sample_size">
        <property name="minimumSize">
         <size>
          <width>64</width>
          <height>0</height>
         </size>
        </property>
        <property name="suffix">
         <string> million points</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>9999</number>
        </property>
        <property name="value">
         <number>2</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_min_big_data_sample_size">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_max_n_columns">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_frame_max_n_columns">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="label_max_n_columns">
        <property name="toolTip">
         <string>This is the maximum number of columns that a dataframe can have in order to be displayed in the visualizer.</string>
        </property>
        <property name="text">
         <string>Maximum number of columns in dataframe  </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBox_max_n_columns">
        <property name="suffix">
         <string>k</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_max_n_columns">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>209</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_general">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QLabel" name="label_visualization">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>32</height>
      </size>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string>Visualization</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="checkBox_setting_enable_1_button_mouse_mode">
     <property name="toolTip">
      <string>When activated, the 1-button mouse mode will be used by default for interacting with the plots, similar to the behaviour in FESA Navigator.</string>
     </property>
     <property name="text">
      <string>Use 1-button mouse mode (Plot)</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="checkBox_setting_enable_downsampling_plots">
     <property name="toolTip">
      <string>Whether or not to enable pyqtgraph's native downsampling when plotting new data. Recommended when visualizing medium-to-large datasets.</string>
     </property>
     <property name="text">
      <string>Use native downsampling by default (Plot)</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="checkBox_setting_display_strings_on_x_axis">
     <property name="toolTip">
      <string>When enabled, x-axis tickers will show original string values instead of numbers, as long as the dataframe index is of type string.</string>
     </property>
     <property name="text">
      <string>Display strings (df.index) as tickers on the x-axis (Plot)</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_ncurves_at_init">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_frame_ncurves_at_init">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="label_ncurves_at_init">
        <property name="toolTip">
         <string>Number of curves that are preloaded when opening the plot panel for the first time.</string>
        </property>
        <property name="text">
         <string>Default number of curves at init (Plot)  </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBox_ncurves_at_init">
        <property name="minimumSize">
         <size>
          <width>64</width>
          <height>0</height>
         </size>
        </property>
        <property name="suffix">
         <string> curves</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>9999</number>
        </property>
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_ncurves_at_init">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_visualization">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QLabel" name="label_timers">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>System Monitor</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="checkBox_setting_enable_system_monitor">
     <property name="toolTip">
      <string>The system monitor plugin can be extremely useful for monitoring the memory usage of large dataframes.</string>
     </property>
     <property name="text">
      <string>Enable system monitor at start-up</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_refresh_rate_system_monitor">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_frame_refresh_rate_system_monitor">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="label_refresh_rate_system_monitor">
        <property name="toolTip">
         <string>The frequency at which the system monitor plugin is updated.</string>
        </property>
        <property name="text">
         <string>Refresh rate  </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinBox_refresh_rate_system_monitor">
        <property name="minimumSize">
         <size>
          <width>64</width>
          <height>0</height>
         </size>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="suffix">
         <string> s</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>1</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_refresh_rate_system_monitor">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>330</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_system_monitor">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QLabel" name="label_colors">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>32</height>
      </size>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Colors</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_color_background">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_frame_color_background">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QPushButton" name="pushButton_color_background">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>16</width>
          <height>16</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16</width>
          <height>16</height>
         </size>
        </property>
        <property name="focusPolicy">
         <enum>Qt::NoFocus</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton{background-color: #000000;}</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_color_background">
        <property name="toolTip">
         <string>Color for the background of all plots.</string>
        </property>
        <property name="text">
         <string>  Background (Plot)</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frame_color_foreground_palette_name">
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_frame_color_foreground_palette_name">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="label_color_foreground_palette_name">
        <property name="toolTip">
         <string>This application utilizes the color_palette function from the Seaborn library. Please refer to the Seaborn documentation to choose the desired palette.</string>
        </property>
        <property name="text">
         <string>Foreground Palette (Plot)  </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="lineEdit_color_foreground_palette_name">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>64</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>colorblind</string>
        </property>
        <property name="maxLength">
         <number>999</number>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_color_foreground_palette_name">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>266</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_colors">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_defaults">
     <property name="topMargin">
      <number>15</number>
     </property>
     <property name="rightMargin">
      <number>5</number>
     </property>
     <property name="bottomMargin">
      <number>5</number>
     </property>
     <item>
      <widget class="QLabel" name="label_saved_changes">
       <property name="font">
        <font>
         <pointsize>9</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel{color:green;}</string>
       </property>
       <property name="text">
        <string>Changes were saved!</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="wordWrap">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_defaults">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_back_to_defaults">
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <property name="text">
        <string>Back to defaults</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_apply">
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <property name="text">
        <string>Apply changes</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
